src/event.c

```diff
@@ -123,6 +123,10 @@ void xmpp_run_once(xmpp_ctx_t *ctx, const unsigned long timeout)
 		if (ret < 0 && !tls_is_recoverable(tls_error(conn->tls))) {
 		    /* an error occured */
 		    conn->error = tls_error(conn->tls);
+		    if( conn->error == 5 )
+			xmpp_debug(ctx, "xmpp", "Unrecoverable TLS write error, syscall %s.", strerror(errno));
```

Comment:
This is useful information, but unfortunately error code `5` can't be parsed here. TLS is implemented as modules (currently openssl, gnutls and windows' schannel). Every module has own error codes. You can implement new interface `tls_strerror()` which returns string representation of the error code.